src/frontend/src/binder/query.rs

```diff
@@ -346,37 +350,52 @@ impl Binder {
                     self.bind_with(with)?;
                 }
 
-                // We assume `left` is base term, otherwise the implementation may be very hard.
-                // The behavior is same as PostgreSQL.
-                // https://www.postgresql.org/docs/16/sql-select.html#:~:text=the%20recursive%20self%2Dreference%20must%20appear%20on%20the%20right%2Dhand%20side%20of%20the%20UNION
-                let bound_base = self.bind_set_expr(*left)?;
+                // todo: to be further reviewed
+                fn gen_query(s: SetExpr) -> Query {
```

Comment:
what if, e.g.,
```sql
with recursive t as
    ((select 1 limit 1)
    union all
    (select n+1 from t as t(n) where n < 5 limit 3));
```
since under current definition of `union`, we can only get `SetExpr` for `left` and `right`.
thus, should we use `BoundSetExpr` as a workaround (and also KISS) at present, or conforming to postgres's behavior?

Reply:
cc @chenzl25, @xiangjinwu.

Reply:
Using `BoundSetExpr` LGTM, because `BoundQuery` and `BoundSetExpr` contains each other actually. 🥵

Reply:
then let's stick with `BoundSetExpr` first.